Tools/clinic/clinic.py

```diff
@@ -208,7 +208,7 @@ def is_legal_py_identifier(s: str) -> bool:
 def ensure_legal_c_identifier(s: str) -> str:
     # for now, just complain if what we're given isn't legal
     if not is_legal_c_identifier(s):
-        fail("Illegal C identifier: {}".format(s))
+        fail(f"Illegal C identifier: {s}")
```

Comment:
Personal preference:
I tend to prefer param style to string interpolation for logging and warning-ish functions. I also find that easier to read for straight-forward stuff like this.

```suggestion
        fail("Illegal C identifier:", s)
```

Reply:
I made similar suggestions to some of the other `fail()`s. Feel free to ignore them.

Reply:
Ah! I forgot that `warn()` and `fail()` took a variadic number of args. Your way is indeed cleaner :)